src/oscar/apps/catalogue/views.py

```diff
@@ -41,6 +41,10 @@ def get(self, request, **kwargs):
         if redirect is not None:
             return redirect
 
+        # Do allow staff members so they can test layout etc.
+        if not product.is_public and not request.user.is_staff:
+            raise Http404(_("Product is not public"))
```

Comment:
Wouldn't a generic 404 be better? No need to expose the fact that a product exists but isn't public.

Reply:
By and large, it is a subject of business logic of each project, not the Oscar framework

Reply:
A simple 404 would do just fine for me

Reply:
feedback applied